### ISSUE TITLE
Block children added to previous block if block is disabled

### ISSUE THREAD

**Issue by Mathias-Syversen** [spicywebau/craft-neo#662]
When disabling a block, if the children are still enabled, the children will appear as children under the previous block.

#### Steps to reproduce
1. Create two blocks
2. Have a child block for example image block on each block.
3. Disable block 2 
4. Image from block 2 appear under block 1

#### Other information
- Neo version: 3.3.9
- Craft version: 4.2.5.1
- Is eager-loading used? Yes

![Screenshot 2022-09-28 at 15 21 51](https://user-images.githubusercontent.com/53942798/192789837-f8a68405-0d52-4575-9745-b429da2775e9.png)
![Screenshot 2022-09-28 at 15 22 06](https://user-images.githubusercontent.com/53942798/192789839-5544b3c7-86cc-4726-82c7-0302bb1c4290.png)

**Comment by vebjorn** [comment]
![neo-debug](https://user-images.githubusercontent.com/4727251/192790586-2996ebf8-da5a-4ece-99ff-ccddca18d3ad.png)
I have the same issue, tried to illustrate it via photoshop, but the situation is the same as for the original poster

**Comment by Mathias-Syversen** [comment]
Main index template for neo:

```{% set blocks = entry.neoField.with([
    "neoSection:background",
    "neoSection:background.media",
    "column:background",
    "column:background.media",
    "logo:logo",
    "mediaGrid:neoMedia",
    "media:neoMedia",
    "people:people",
    "people:people.profilePicture",
    "slider:neoImages",
    "form:form",
    "map:logo"
    ]).collect() %}

{% for block in blocks %}
    {% do block.useMemoized(blocks) %}
{% endfor %}

{% for block in blocks %}
    {% if block.level == 1 %}
        {% include["_components/neo/blocks/" ~ block.type,
        "_components/matrix/neo/default"] with { block: block } %}
    {% endif %}
{% endfor %}```

And inside the column template:

```    {% for child in block.children %}
        {% include "_components/neo/blocks/" ~ child.type with { block: child } %}
    {% endfor %}```

**Comment by ttempleton** [comment]
Thanks for reporting that - should be fixed now in Neo 3.4.0.
